src/test_util.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(test)]
+
+use std::fmt::Debug;
+
+/// Test that two values are equal, with a better error than `assert_eq`
+pub fn assert_equal<A>(expected_value: &A, actual_value: &A)
```

Comment:
Diff output: super useful with our hard to read results 👍. No need to change anything since you've done the work, but if we ever want _fancy color output_, [pretty-assertions](https://github.com/colin-kiegel/rust-pretty-assertions) seemed to work nicely.

Reply:
I think I tried `pretty-assertions` and found that it was not giving me reasonable diffs (iirc, it was not handling `\n` correctly, so I just got all `-` rows). Perhaps I was doing something silly though.

Reply:
Oh no, I was trying some other crate. That *does* look nice. However, maybe i'll still with this, because I was hoping to change from using `Debug` output to something else that would let us give names to the values (not just the interned indices)

Reply:
Yes, the `debug_with` plan ! Sounds good.